api/filter-alerts.go

```diff
@@ -0,0 +1,228 @@
+		labels[i] = graphql.String(label)
+	}
+
+	updateAlert := humiographql.UpdateFilterAlert{
+		ViewName:           humiographql.RepoOrViewName(viewName),
+		ID:                 graphql.String(newFilterAlert.ID),
+		Name:               graphql.String(newFilterAlert.Name),
+		Description:        graphql.String(newFilterAlert.Description),
+		QueryString:        graphql.String(newFilterAlert.QueryString),
+		ActionIdsOrNames:   actionsIdsOrNames,
+		Labels:             labels,
+		Enabled:            graphql.Boolean(newFilterAlert.Enabled),
+		RunAsUserID:        graphql.String(newFilterAlert.RunAsUserID),
+		QueryOwnershipType: humiographql.QueryOwnershipType(newFilterAlert.QueryOwnershipType),
+	}
+
+	variables := map[string]interface{}{
+		"input": updateAlert,
+	}
+
+	err := fa.client.Mutate(&mutation, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	filterAlert := mapHumioGraphqlFilterAlertToFilterAlert(mutation.FilterAlert)
+
+	return &filterAlert, nil
+}
+
+func (fa *FilterAlerts) Add(viewName string, newFilterAlert *FilterAlert) (*FilterAlert, error) {
+	if newFilterAlert == nil {
+		return nil, fmt.Errorf("newFilterAlert must not be nil")
+	}
+
+	var mutation struct {
+		humiographql.FilterAlert `graphql:"createFilterAlert(input: $input)"`
+	}
+
+	actionsIdsOrNames := make([]graphql.String, len(newFilterAlert.Actions))
+	for i, actionIdOrName := range newFilterAlert.Actions {
+		actionsIdsOrNames[i] = graphql.String(actionIdOrName)
+	}
+
+	labels := make([]graphql.String, len(newFilterAlert.Labels))
+	for i, label := range newFilterAlert.Labels {
+		labels[i] = graphql.String(label)
+	}
+
+	createFilterAlert := humiographql.CreateFilterAlert{
+		ViewName:           humiographql.RepoOrViewName(viewName),
+		Name:               graphql.String(newFilterAlert.Name),
+		Description:        graphql.String(newFilterAlert.Description),
+		QueryString:        graphql.String(newFilterAlert.QueryString),
+		ActionIdsOrNames:   actionsIdsOrNames,
+		Labels:             labels,
+		Enabled:            graphql.Boolean(newFilterAlert.Enabled),
+		RunAsUserID:        graphql.String(newFilterAlert.RunAsUserID),
+		QueryOwnershipType: humiographql.QueryOwnershipType(newFilterAlert.QueryOwnershipType),
+	}
+
+	variables := map[string]interface{}{
+		"input": createFilterAlert,
+	}
+
+	err := fa.client.Mutate(&mutation, variables)
+	if err != nil {
+		return nil, err
+	}
+
+	filterAlert := mapHumioGraphqlFilterAlertToFilterAlert(mutation.FilterAlert)
+
+	return &filterAlert, nil
+}
+
+func (fa *FilterAlerts) Get(viewName, filterAlertName string) (*FilterAlert, error) {
+	filterAlerts, err := fa.List(viewName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list filter alerts: %w", err)
+	}
+	for _, filterAlert := range filterAlerts {
+		if filterAlert.Name == filterAlertName {
+			return &filterAlert, nil
+		}
+	}
+
+	return nil, FilterAlertNotFound(filterAlertName)
+}
+
+func (fa *FilterAlerts) Delete(viewName, filterAlertName string) error {
```

Comment:
The GraphQL mutation takes an id, at least.

Reply:
As @jespermbyskov said above. The cli is focused on names, not IDs. So I believe this filtering delete and get is the right way to go.

Reply:
I agree, the cli is focused on names. But this is part of the API package, which can be/is used outside of the cli commands.

Reply:
All the other APIs also use name for deletion. I also think that it would be better to use IDs, but since that is not how this API is designed, I think that we should follow the "convention".

Reply:
I know the other APIs are "bad" as well, I don't think that should stop us from doing the right thing in new APIs. To follow the existing API a "DeleteByName" method could be created that does `List` + `Delete`.

I think the API design stems from a point where everything in Humio was referenced by name, not ID.

Reply:
But maybe we should hear @SaaldjorMike, his team are probably most likely to be users of this API.

Reply:
> I think the API design stems from a point where everything in Humio was referenced by name, not ID.
Yes. Back in the day the "ID" was the name itself. Since then we've added real ID's to the entities, and I agree with @danielamkaer that just because the other parts of the API is bad, doesn't mean we should continue to add more things that follows the same approach. Throwing in a "DeleteByName" sounds good to me, then it's still easy to use this in a similar way to the rest of our resource types.